src/main/java/net/smartcosmos/edge/things/service/GetThingEdgeServiceDefault.java

```diff
@@ -139,8 +154,39 @@ public GetThingEdgeServiceDefault(
         return buildForwardingResponse(thingResponse);
     }
 
+    private ResponseEntity<?> getThingsMergeMetadataUnpaged(
+        String type,
+        Map<String, Set<String>> urns,
+        ResponseEntity thingResponse,
+        SmartCosmosUser user) {
+
+        if (thingResponse.hasBody() && thingResponse.getBody() instanceof RestUnpagedThingResponse) {
+
+            RestUnpagedThingResponse things = (RestUnpagedThingResponse) thingResponse.getBody();
+            try {
+                List<Map<String, Object>> data = collectFindByTypeData(things.getData(), null, user);
+
+                RestEdgeThingResponseDto<Map<String, Object>> responsePage = RestEdgeThingResponseDto.<Map<String, Object>>builder()
+                    .data(data)
+                    .build();
+
+                return ResponseEntity.ok()
+                    .contentType(MediaType.APPLICATION_JSON_UTF8)
+                    .body(responsePage);
+            } catch (RestException e) {
+                String msg = getByTypeAndUrnsLogMessage(type, urns, user, e.toString());
+                log.error(msg);
+                log.debug(msg, e);
+                return e.getResponseEntity();
```

Comment:
This could cause issues because header fields are duplicated: The original ones are kept and new ones are added. This is the reason why `buildForwardingResponse()` is used in all the other places.